.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,25 @@
+name: test-all
+
+on: [pull_request]
+
+jobs:
+  test-all:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run on NodeJS version ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: |
+        npm install
+        npm run build
+        npm run test
+        npm run lint:test
```

Comment:
maybe start wit the lint - it is lightweght so when it fails, it saves time by not going into build/test stages

Reply:
That's actually clever :)